### PR TITLE
feat: upgrade to be compatible with Dart 3

### PR DESCRIPTION
**Pull request by prokopsimek**
Screenshot from built application for MacOS in Dart SDK version: 3.0.7 (stable) (Mon Jul 24 13:17:56 2023 +0000) on "macos_x64".

Screenshot 2023-08-11 at 15 53 05

### CODE REVIEW

**Comment by ViolanteCodes** [comment]
@jakelumetta it doesn't look like we've added versioning in for the dart SDK yet - do you want the current version of it tagged as v1 and then have a version bump for this PR, as this is a pretty big change?

**Comment by jakelumetta** [comment]
@ViolanteCodes Hmm yeah - not sure why we don't have versioning on this yet but we definitely should. Sounds good to me. Can we possibly tag what's currently released as v1 and then tag this new PR as 1.1 or 2?

**Review by ViolanteCodes — CHANGES_REQUESTED** [review]
@prokopsimek we talked about how to handle the dart v3 upgrade as far as versioning, as our previous version number was 0.3.0, which implies it's still in beta development.

Can we have this PR bring the release to 1.0.0, and I'll push this out as initial release for dart 3 compatibility?

**Comment by ViolanteCodes** [comment]
> @ViolanteCodes Hmm yeah - not sure why we don't have versioning on this yet but we definitely should. Sounds good to me. Can we possibly tag what's currently released as v1 and then tag this new PR as 1.1 or 2?

@jakelumetta 

After some more poking around, I determined what we currently have out already had a version number, 0.3.0, inside of pubspec.yaml but not github. 

I didn't want to manually edit the pubspec.yaml to tag what we had out as version 1.0.0, as I wasn't sure if that would require changes to the lock files, etc., and I don't have the architecture on my system to run this package (and people are already using the current 0.3.0)

As such, I tagged what we currently had in github as 0.3.0 to match what was in the pubspec.yaml (especially since people are already using it with that version), and am requesting agency tag the dart 3 upgrades (which are backwards compatible), as v1.0.0, just so everything stays clean and easy to follow. 

Let me know if that's an issue - if so, i'll see if I can't get the current pubspec.yaml updated to v1.0.0, and then have the agency move the dart 3 PR to v1.1.1.

File: pubspec.yaml [hunk 1]
```diff
@@ -1,19 +1,19 @@
 name: buttercms_dart
 description: The official ButterCMS package for Flutter. Use the full feature set Butter offers, now available on Flutter.
-version: 0.3.0
+version: 0.4.0
```
Comment (prokopsimek) [inline pubspec.yaml]:
```suggestion
version: 1.0.0
```

**Review by ViolanteCodes — APPROVED** [review]
[Per printout, ](https://github.com/ButterCMS/buttercms-dart/pull/7/files#diff-8b7e9df87668ffa6a04b32e1769a33434999e54ae081c52e5d943c541d4c0d25L10) - should be backwards compatible.

**Pull request merged**